plugins/embed-optimizer/hooks.php

```diff
@@ -48,7 +48,7 @@ function embed_optimizer_filter_oembed_html( string $html ): string {
 				}
 			}
 		} elseif ( 'SCRIPT' === $html_processor->get_tag() ) {
-			if ( ! $html_processor->get_attribute( 'src' ) ) {
+			if ( ! is_string( $html_processor->get_attribute( 'src' ) ) ) {
```

Comment:
We can avoid the `is_string` call and confirm that the result is not `null`.

```suggestion
			if ( null !== $html_processor->get_attribute( 'src' ) ) {
```

🔢

Reply:
The return value of `get_attribute()` is either `string`, `null`, or `true`.  So if we just checked if not `null` then this could actually be a case where `<script src>` which wouldn't be right. This is a script which isn't inline but also its external URL is void so it doesn't make any request. So that's somewhat of a unique case, unlikely to be encountered. So there are three possible states:

1. `<script src="script.js"></script>`
2. `<script>document.write('hi');</script>`
3. `<script src></script>`

The third case is not treated like either an inline script or an external script. But since it doesn't make sense for Embed Optimizer to try to lazy-load a void `src`, then I think it makes sense to treat it as if it is an inline script.